lib/ddq-backend-mysql.js

```diff
@@ -331,8 +331,10 @@ module.exports = (configValidation, crypto, EventEmitter, mysql, timers) => {
         /**
          * Resets the relevant flags and clears the timers, which stops both
          * polling and restoring.
+         *
+         * @param {Function} [callback]
          */
-        pausePolling() {
+        stopListening(callback) {
```

Comment:
We talked a bit about this.  I think I'll discuss it more with you, but the essence of the solution is to, I believe, track a few more things in DDQ and pull more logic out of the plugins.  Let DDQ coordinate the callbacks for starting/stopping/etc.  The plugin shouldn't even worry about `this.currentlyRestoring` nor `this.currentlyPolling` in the way it's currently approaching that problem.  It should track when it is doing an operation, but probably not when it is stalling.